src/sync.rs

```diff
@@ -62,7 +62,7 @@ struct Mutex {
     /// The queue of threads waiting for this mutex.
     queue: VecDeque<ThreadId>,
     /// Data race handle
-    data_race: DataRaceLockHandle
+    data_race: VClock
```

Comment:
In the other file you have these nice comments explaining what these clocks represent. Can you do the same here (also for the other things in this file)? I think it'd be something like "the clock of the thread that last released the mutex", or so?